src/main/java/io/github/thebusybiscuit/sensibletoolbox/core/gui/STBInventoryGUI.java

```diff
@@ -197,10 +197,12 @@ public void show(Player player) {
                 monitor.doRepaint();
             }
         }
-        Debugger.getInstance().debug(player.getName() + " opened GUI for " + getOwningItem());
-        setOpenGUI(player, this);
-        listener.onGUIOpened(player);
-        player.openInventory(inventory);
+        if (getOpenGUI(player) == null) {
+            Debugger.getInstance().debug(player.getName() + " opened GUI for " + getOwningItem());
+            setOpenGUI(player, this);
+            listener.onGUIOpened(player);
+            player.openInventory(inventory);
+        }
```

Comment:
A private helper method for the if-statement might make this easier to read.
`private boolean hasOpenGUI(player)` for example